platform/os_windows.go

```diff
@@ -118,3 +120,59 @@ func KillProcessByName(processName string) {
 	cmd := fmt.Sprintf("taskkill /IM %v /F", processName)
 	ExecuteCommand(cmd)
 }
+
+// CreateExtHnsNetwork creates ext HNS network if not present.
+func CreateExtHnsNetwork(createExtNetworkType string) error {
+	if len(strings.TrimSpace(createExtNetworkType)) == 0 {
+		return nil
+	}
+
+	if !strings.EqualFold(createExtNetworkType, "l2bridge") &&
+		!strings.EqualFold(createExtNetworkType, "l2tunnel") {
+		return fmt.Errorf("Invalid ext hns network type %s", createExtNetworkType)
+	}
+
+	log.Printf("[Azure CNS] CreateExtHnsNetwork")
+	extHnsNetwork, err := hcsshim.GetHNSNetworkByName("ext")
+
+	if extHnsNetwork != nil {
+		log.Printf("[Azure CNS] Found existing ext hns network with type: %s", extHnsNetwork.Type)
+		if !strings.EqualFold(createExtNetworkType, extHnsNetwork.Type) {
```

Comment:
should we delete the the network and recreate it with the type passed in?

Reply:
I don't want to do that because the whole vm might have been set up with a different type ( it may have other hnsnetworks with that type ). This would error out early so that the client / orchestrator can device what needs to be done.